### ISSUE TITLE
Problem reusing batchnorm

### ISSUE THREAD

**Issue by bernardohenz** [fchollet/keras#3374]
I am trying to separate my full model into two parts (similar to autoencoder having encoder/decoder), but I keep getting problems when doing so. I wrote a simple case to reproduce my problem, first I started by trying doing so:

``` python
from keras.layers import Input, Dense, Convolution2D, MaxPooling2D, UpSampling2D,merge
from keras.models import Model,Sequential
from PIL import Image
import numpy as np
from keras.layers.normalization import BatchNormalization

main_input= Input(shape=(3, 64, 64))

#first 'model'
conv1 = Convolution2D(64, 3, 3, border_mode='same')(main_input)
conv1 = BatchNormalization(mode=0, axis=1)(conv1)
conv1 = Convolution2D(128, 3, 3, border_mode='same')(conv1)
conv1 = BatchNormalization(mode=0, axis=1)(conv1)
conv1 = Convolution2D(256, 3, 3, border_mode='same')(conv1)

#second 'model'
conv2 = Convolution2D(256, 3, 3, border_mode='same')(conv1)
conv2 = BatchNormalization(mode=0, axis=1)(conv2)
conv2 = Convolution2D(128, 3, 3, border_mode='same')(conv2)
conv2 = BatchNormalization(mode=0, axis=1)(conv2)
conv2 = Convolution2D(64, 3, 3, border_mode='same')(conv2)
conv2 = BatchNormalization(mode=0, axis=1)(conv2)
conv2 = Convolution2D(3, 3, 3, border_mode='same')(conv2)

first_model = Model(input=main_input,output=conv1)
second_model = Model(input=conv1,output=conv2)
full_model = Model(input=main_input,output=conv2)
```

But when trying to create de **second_model** I get the error _'Exception: Graph disconnected: cannot obtain value for tensor input_1 at layer "input_1". The following previous layers were accessed without issue: []'_

Then, I tried to get layer by layer of the full model:

``` python
full_model = Model(input=main_input,output=conv2)

second_input= Input(shape=(256, 64, 64))
current_node = second_input
for i in range(len(first_model.layers),len(full_model.layers)):
    current_layer = full_model.layers[i]
    current_node = current_layer(current_node)

second_model = Model(input=second_input,output=current_node)
```

However I got the error _'Exception: You are attempting to share a same `BatchNormalization` layer across different data flows. This is not possible. You should use `mode=2` in `BatchNormalization`, which has a similar behavior but is shareable (see docs for a description of the behavior).'_

Could someone help me?
Obs: I am building my architecture using Graph blocks, so I prefer if the solution can be done in a similar way

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [X] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [X] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [X] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by liquidscience** [comment]
@bernardohenz I was having a similar issue , were you able to resolve this?

**Comment by bernardohenz** [comment]
@liquidscience What I did was to train the entire network, save it, and when I would like to use only "a part" of it, I load it , and remove the layers I don't want.
OBS: Do not forget to update the output params when deleting layers (see [https://github.com/fchollet/keras/issues/3442#issuecomment-255626352](https://github.com/fchollet/keras/issues/3442#issuecomment-255626352))
